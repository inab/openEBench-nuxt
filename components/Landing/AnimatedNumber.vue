<template>
	<v-progress-circular v-if="value == 0" indeterminate></v-progress-circular>
	<span v-else v-observe-visibility="visibilityChanged">{{
		displayValue
	}}</span>
</template>

<script>
import gsap from 'gsap';
import { ObserveVisibility } from 'vue-observe-visibility';

export default {
	name: 'App',
	directives: {
		ObserveVisibility,
	},
	props: { value: { type: Number, default: 0 } },
	data() {
		return {
			inputValue: 0,
			displayValue: 0,
			tweenValue: 0,
			isVisible: false,
		};
	},
	watch: {
		inputValue() {
			gsap.to(this, {
				tweenValue: this.inputValue,
				onUpdate: () => {
					this.displayValue = Math.ceil(this.tweenValue);
				},
				duration: 2,
			});
		},
	},
	methods: {
		visibilityChanged(isVisible) {
			if (!this.isVisible && isVisible) {
				this.isVisible = isVisible;
				this.inputValue = this.value;
			} else {
				this.isVisible = false;
				this.inputValue = 0;
			}
		},
	},
};
</script>
