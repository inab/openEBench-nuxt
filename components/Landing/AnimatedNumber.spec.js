import { mount } from '@vue/test-utils';
import AnimatedNumber from './AnimatedNumber.vue';

const factory = (propsData) => {
	return mount(AnimatedNumber, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
		propsData,
	});
};

describe('AnimatedNumber', () => {
	it('is instantiated', () => {
		const wrapper = factory({
			value: 0,
		});
		expect(wrapper).toBeTruthy();
	});

	it('should show a progress spinner if value == 0', () => {
		const wrapper = factory({
			value: 0,
		});
		expect(wrapper.find('.v-progress-circular').exists()).toBe(true);
	});

	it('should show the number if value > 0', () => {
		const wrapper = factory({
			value: 999,
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should update the local props @visibility change accordingly', () => {
		const wrapper = factory({
			value: 10,
		});
		expect(wrapper.vm.inputValue).toBe(0);
		expect(wrapper.vm.isVisible).toBe(false);

		wrapper.vm.visibilityChanged(true);

		expect(wrapper.vm.inputValue).toBe(10);
		expect(wrapper.vm.isVisible).toBe(true);

		wrapper.vm.visibilityChanged(false);

		expect(wrapper.vm.inputValue).toBe(0);
		expect(wrapper.vm.isVisible).toBe(false);
	});
});
