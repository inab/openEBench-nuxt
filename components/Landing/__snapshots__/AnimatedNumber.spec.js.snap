// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedNumber should show the number if value > 0 1`] = `"<span>0</span>"`;
